jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -547,13 +550,12 @@ def extract(self, return_bg=False, add_data=True, **kwargs):
         spec.meta['_pixel_scale_factor'] = pix_scale_factor
 
         # inform the user if scale factor keyword not in metadata
-        if 'PIXAR_SR' not in self.cube.meta:
-            snackbar_message = SnackbarMessage(
+        if _first_run and ('PIXAR_SR' not in self.cube.meta):
```

Comment:
I fixed the snackbar spam here because it is too annoying to review with the bug in this PR.

This is a "quick and dirty" fix. Is it worrisome that this line is called so many times? Probably but not something I want to fix as a reviewer of this PR.

Reply:
We have a plan to address this as well, so will need to revert this full commit, is it needed to have until then?

Reply:
I think this temporary fix should go in because otherwise the snackbar is drowned out and I cannot see the actually useful  messages.

Reply:
i dont know if it makes sense to have the snackbar only pop up when you open the extraction plugin, since an extraction is done before that, but if this is temporary then thats fine

Reply:
Another option is to just temporary comment this out entirely until someone can fix it properly. (I actually think it should only emit once at load time, not here.) As it is right now, it is doing more harm than good.

Reply:
this doesn't seem to be too related to the scope of this PR, but if its absolutely necessary to get rid of here instead of in a separate PR, then I would vote to have it just removed (since that is the eventual end result) rather than introducing new code that we know we'll be getting rid of in an upcoming effort

Reply:
Another question is why is this code invoked so many times? Is it oversubscribing to events? Are we doing unnecessary re-calculations? But that is, again, out of scope here.

Reply:
If this really bothers you, I can undo these changes (or Clare can). I can open up a separate PR.

Reply:
OK I undid the changes here. I think Kyle said he is going to look into it separately.